client/question/services/EventHandlerService.js

```diff
@@ -19,18 +19,58 @@
 tie.factory('EventHandlerService', [
   '$http', 'ServerHandlerService', function($http, ServerHandlerService) {
 
+    /**
+     * Global object to keep track of the current batch of events to send.
+     * @type [Event]
+     */
+    var _currentEventBatch = [];
+
+    /**
+     * Submits the current batch of event data to TIE's backend.
+     *
+     */
+    var sendCurrentEventBatch = function() {
+      if (getCurrentEventBatchLength() === 0) {
+        // No point in sending an empty batch.
+        return null;
+      }
+      var data = {
+        events: _currentEventBatch,
+        timeSentToBackendMsec: (new Date()).getTime()
+      };
+      return $http.post('/ajax/event/send_event_batch', data).then(
+        function() {
+          _currentEventBatch.length = 0;
+        }, function() {
+          // Otherwise, we should do nothing.
+          // Since the call errored, don't dump the current batch.
+      });
+    };
+
+    var getCurrentEventBatchLength = function() {
+      return _currentEventBatch.length;
+    };
+
     return {
+      getCurrentEventBatchLength: getCurrentEventBatchLength,
```

Comment:
In the previous comment, I meant that you could prefix the key with an underscore: _getCurrentEventBatchLength

Reply:
Ah, okay. Done.